test/sqllogictest/aclitem.slt

```diff
@@ -0,0 +1,203 @@
+=r/test_role
+
+query T
+SELECT mz_internal.make_mz_aclitem('p', 'u2', 'INSERT, SELECT')::text
+----
+=ar/test_role
+
+query error mz_aclitem grantor cannot be PUBLIC role
+SELECT mz_internal.make_mz_aclitem('u1', 'p', 'CREATE')
+
+query error mz_aclitem grantor cannot be PUBLIC role
+SELECT mz_internal.make_mz_aclitem('u1', 'p', 'CREATE, USAGE')
+
+query T
+SELECT mz_internal.mz_aclitem_grantee(mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE'))
+----
+u1
+
+query T
+SELECT mz_internal.mz_aclitem_grantee(mz_internal.make_mz_aclitem('p', 'u2', 'DELETE'))
+----
+p
+
+query T
+SELECT mz_internal.mz_aclitem_grantor(mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE'))
+----
+u2
+
+query T
+SELECT mz_internal.mz_aclitem_privileges(mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE'))
+----
+d
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE') = mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+true
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE') <> mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+false
+
+query B
+SELECT mz_internal.make_mz_aclitem('u3', 'u2', 'DELETE') = mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+false
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u3', 'DELETE') = mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+false
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u3', 'DELETE, USAGE') = mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+false
+
+query B
+SELECT mz_internal.make_mz_aclitem('u3', 'u2', 'DELETE') <> mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+true
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u3', 'DELETE') <> mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+true
+
+query B
+SELECT mz_internal.make_mz_aclitem('u1', 'u3', 'DELETE, USAGE') <> mz_internal.make_mz_aclitem('u1', 'u2', 'DELETE')
+----
+true
+
+# RoleIds that don't exist are printed as IDs instead of names
+
+query T
+SELECT mz_internal.make_mz_aclitem('u99991', 'u2', 'CREATE')::text
+----
+u99991=C/test_role
+
+query T
+SELECT mz_internal.make_mz_aclitem('u1', 'u87398', 'CREATE')::text
+----
+materialize=C/u87398
+
+query T
+SELECT mz_internal.make_mz_aclitem('u3251', 's345', 'CREATE')::text
+----
+u3251=C/s345
+
+# Test parsing errors
+
+query error couldn't parse role id 'uasdf7890ad'
+SELECT mz_internal.make_mz_aclitem('u1', 'uasdf7890ad', 'CREATE')
+
+query error couldn't parse role id 'sd98fas9df8'
+SELECT mz_internal.make_mz_aclitem('sd98fas9df8', 's1', 'CREATE')
+
+query error unrecognized privilege type: "asdfa ljefioj"
+SELECT mz_internal.make_mz_aclitem('u1', 's1', 'asdfa ljefioj')
+
+# Test mz_acl_item_contains_privilege NULLs
+
+query B
+SELECT mz_internal.mz_acl_item_contains_privilege(NULL, 'SELECT')
+----
+NULL
+
+query B
+SELECT mz_internal.mz_acl_item_contains_privilege(mz_internal.make_mz_aclitem('u1', 'u2', 'CREATE'), NULL)
+----
+NULL
+
+query B
+SELECT mz_internal.mz_acl_item_contains_privilege(NULL, NULL)
+----
+NULL
```

Comment:
This entire sections was moved verbatim from `privilege_grants.slt`, since that file was getting too big.